plugins/plugin-pullrequest-parent/che-plugin-pullrequest-ide/src/main/java/org/eclipse/che/plugin/pullrequest/client/steps/GenerateReviewFactoryStep.java

```diff
@@ -35,30 +39,45 @@
 
 import static org.eclipse.che.ide.api.notification.StatusNotification.DisplayMode.NOT_EMERGE_MODE;
 import static org.eclipse.che.ide.api.notification.StatusNotification.Status.FAIL;
+import static org.eclipse.che.plugin.pullrequest.shared.Constants.CHE_PULLREQUEST_GENERATE__REVIEW__FACTORY;
 
 /**
  * Generates a factory for the contribution reviewer.
  */
 @Singleton
 public class GenerateReviewFactoryStep implements Step {
-    private final ContributeMessages   messages;
-    private final AppContext           appContext;
-    private final NotificationManager  notificationManager;
-    private final FactoryServiceClient factoryService;
+    private final ContributeMessages               messages;
+    private final AppContext                       appContext;
+    private final NotificationManager              notificationManager;
+    private final FactoryServiceClient             factoryService;
+    private final PullRequestWorkflowServiceClient pullRequestWorkflowService;
 
     @Inject
     public GenerateReviewFactoryStep(final ContributeMessages messages,
                                      final AppContext appContext,
                                      final NotificationManager notificationManager,
+                                     final PullRequestWorkflowServiceClient pullRequestWorkflowService,
                                      final FactoryServiceClient factoryService) {
         this.messages = messages;
         this.appContext = appContext;
         this.notificationManager = notificationManager;
+        this.pullRequestWorkflowService = pullRequestWorkflowService;
         this.factoryService = factoryService;
     }
 
     @Override
     public void execute(final WorkflowExecutor executor, final Context context) {
+        pullRequestWorkflowService.getSettings() //
+                                  .then((Map<String, String> settings) -> {
+                                      if (!Boolean.parseBoolean(settings.getOrDefault(CHE_PULLREQUEST_GENERATE__REVIEW__FACTORY, "true"))) {
```

Comment:
I think we don't need a map there, we can inject a property with name "che.pullrequest.generate_review_factory"